File: offer/medium/PHP/cuttingRope2.php
<?php

// 剑指 Offer 14- II. 剪绳子

function cuttingRope2($n)
{
    if ($n < 4) {
        return $n - 1;
    }

    $result = 1;
    while ($n > 4) {
        $result *= 3;
        $n -= 3;
    }
    return $result * $n;
}

echo cuttingRope2(5);

File: offer/medium/Golang/cuttingRope2.go
package main

import "fmt"

func main() {

	n := 100000000000000
	res := cuttingRope2(n)
	fmt.Println(res)
}

func cuttingRope2(n int) int {
	if  n < 4 {
		return n - 1
	}

	result := 1
	for n > 4 {
		result = (result * 3) % 1000000007
		n -= 3
	}
	return (result * n) % 1000000007
}
